src/main/java/datawave/query/util/MetadataHelper.java

```diff
@@ -1070,32 +1062,82 @@ public long getCardinalityForField(String fieldName, Date begin, Date end) throw
     public long getCardinalityForField(String fieldName, String datatype, Date begin, Date end) throws TableNotFoundException {
         log.trace("getCardinalityForField from table: " + metadataTableName);
         Text row = new Text(fieldName.toUpperCase());
+        FrequencyFamilyCounter dateFreqMap = new FrequencyFamilyCounter();
         
         // Get all the rows in DatawaveMetadata for the field, only in the 'f'
-        // colfam
+        // colfamily
         Scanner bs = ScannerHelper.createScanner(connector, metadataTableName, auths);
         
         Key startKey = new Key(row);
         bs.setRange(new Range(startKey, startKey.followingKey(PartialKey.ROW)));
         bs.fetchColumnFamily(ColumnFamilyConstants.COLF_F);
         
         long count = 0;
+        Value aggregatedValue = null;
+        
+        String cq = "";
         
         for (Entry<Key,Value> entry : bs) {
-            Text colq = entry.getKey().getColumnQualifier();
+            if (entry.getKey().getColumnQualifier().toString().startsWith(COL_QUAL_PREFIX))
```

Comment:
Since we are dealing with backward compatibility, we cannot assume there will be only one aggregated value, nor can we assume that all of the key/values contain aggregated values.  We need to change this to aggregate across all of the key/values returned from the scanner.

Reply:
If we are to assume that we have only aggregated values, then why do we have a for loop here.  We can simply seek and get the next value.

Reply:
I think I accomplished this in the last commit.